shell.go

```diff
@@ -35,23 +40,26 @@ func WithCommand(cmd string) func(*shellOptions) {
 // session is started. This is a synchronous, blocking call.
 //
 // Can be used with optional helper parameters UseCWD and WithCommand.
-func (d *Distro) Shell(opts ...func(*shellOptions)) (err error) {
+//
+// Return values:
+// - exitCode is guaranteed to be usable, even in case of error, so checking for
+// exitCode != 0 is a valid way of ensuring a command went through.
+// - err will report problems launching the shell, but not errors on Linux's side.
```

Comment:
I would say "not errors on the executed command inside WSL itself"

Reply:
Btw, are we really sure we want this behaviour? Can it be rather typed error?

I feel it weird if I compare to os/exec.Run() for instance:
`The returned error is nil if the command runs, has no problems copying stdin, stdout, and stderr, and exits with a zero exit status.`

I would expect to check for error, then, getting the exit code if I want to know more (depending on the error typed maybe. wrapped in WindowsError and CommandError for instance)

Reply:
Ideally I'd be able to use `exec.ExitError`; but that requires finding the pid of the process that failed (which is no longer active, because it failed).

This implementation mimicks the WSL api, but it's definitely up for discussion.

Reply:
Anyways I decided to always return a `ShellError` if the exit code is not 0.

Reply:
You are being reasonable after a good night of sleep \o/